DeusEx/Classes/Player.uc

```diff
@@ -0,0 +1,11 @@
+class DXRPlayer merges DeusExPlayer;
+
+function bool AddInventory( inventory NewItem )
+{
+    local DXRBannedItems ban_items;
+
+    foreach AllActors(class'DXRBannedItems', ban_items) {
+        if ( ban_items.ban(self, NewItem) ) return true;
```

Comment:
I could also do something like this to prevent excessive melee weapons, or maybe enemies should just drop melee weapons and nanokeys on the ground instead of keeping them in the corpse